src/hooks/useENSSearch.ts

```diff
@@ -61,26 +89,30 @@ export default function useENSSearch({
         valid: true,
       };
     } else {
-      // we need the expiration and registration date when is not available
-      const registrationDate = await fetchRegistrationDate(name + ENS_DOMAIN);
-      const nameExpires = await getNameExpires(name);
+      if (errors[2]) throw errors[2];
```

Comment:
can we not just do it the original way, but place the `fetchRegistrationDate` and `getNameExpires` in a `Promise.all`?